services/thermal_mgr/thermal_mgr.c

```diff
@@ -42,20 +44,71 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 }
 
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
-  /* Send an event to the thermal manager queue */
-
-  return ERR_CODE_SUCCESS;
+    // Check if the event pointer is valid
+    if (event == NULL) {
+        return ERR_CODE_INVALID_ARG;  
+    }
+
+    // Check if the thermal manager queue handle is valid
+    if (thermalMgrQueueHandle == NULL) {
+        return ERR_CODE_INVALID_STATE;  
+    }
+
+    // Attempt to send the event to the thermal manager queue
+    if (xQueueSend(thermalMgrQueueHandle, event, THERMAL_MGR_QUEUE_TIMEOUT) == pdPASS) {
+        return ERR_CODE_SUCCESS;  
+
+    // Return error if sending the event fails
+    return ERR_CODE_QUEUE_FULL;  
+    }
```

Comment:
Based on the curly braces it looks as tho this is also inside the success case for the xQueueSend so move the return below the }